src/librustc_typeck/check/writeback.rs

```diff
@@ -744,20 +738,19 @@ impl<'cx, 'tcx> WritebackCx<'cx, 'tcx> {
         }
     }
 
-    fn resolve<T>(&self, x: &T, span: &dyn Locatable) -> T::Lifted
+    fn resolve<T>(&self, x: &T, span: &dyn Locatable) -> T
     where
-        T: TypeFoldable<'tcx> + ty::Lift<'tcx>,
+        T: TypeFoldable<'tcx>,
     {
         let x = x.fold_with(&mut Resolver::new(self.fcx, span, self.body));
-        if let Some(lifted) = self.tcx().lift_to_global(&x) {
-            lifted
-        } else {
+        if cfg!(debug_assertions) && x.has_local_value() {
```

Comment:
Is it really that expensive? Anyway it should be impossible, given a correct `Resolver`.

Reply:
I don't know how expensive these are. They're not free though, so I'd prefer to not have them for release builds.

And yes, it should be impossible, that's why it's an assertion =P